src/srcShared/fs.ts

```diff
@@ -0,0 +1,133 @@
+    }
+
+    async appendFile(path: Uri | string, content: Uint8Array | string): Promise<void> {
+        path = FileSystemCommon.getUri(path)
+
+        const currentContent: Uint8Array = (await this.fileExists(path)) ? await this.readFile(path) : new Uint8Array(0)
+        const currentLength = currentContent.length
+
+        const newContent = FileSystemCommon.asArray(content)
+        const newLength = newContent.length
+
+        const finalContent = new Uint8Array(currentLength + newLength)
+        finalContent.set(currentContent)
+        finalContent.set(newContent, currentLength)
+
+        return this.writeFile(path, finalContent)
+    }
+
+    async fileExists(path: Uri | string): Promise<boolean> {
+        path = FileSystemCommon.getUri(path)
+        const stat = await this.stat(path)
+        return stat === undefined ? false : stat.type === vscode.FileType.File
+    }
+
+    async writeFile(path: Uri | string, data: string | Uint8Array): Promise<void> {
+        path = FileSystemCommon.getUri(path)
+        return fs.writeFile(path, FileSystemCommon.asArray(data))
+    }
+
+    /**
+     * @returns the stat of the file or undefined if it does not exist
+     */
+    async stat(uri: vscode.Uri | string): Promise<vscode.FileStat | undefined> {
+        const path = FileSystemCommon.getUri(uri)
+        try {
+            return await fs.stat(path)
+        } catch (err) {
+            if (err instanceof vscode.FileSystemError && err.code === 'FileNotFound') {
+                return undefined
+            }
+            throw err
+        }
+    }
+
+    async delete(uri: vscode.Uri | string): Promise<void> {
+        const path = FileSystemCommon.getUri(uri)
+        return fs.delete(path, { recursive: true })
+    }
+
+    async readdir(uri: vscode.Uri | string): Promise<[string, vscode.FileType][]> {
+        const path = FileSystemCommon.getUri(uri)
+        return await fs.readDirectory(path)
+    }
+
+    // -------- private methods --------
+    static readonly #decoder = new TextDecoder()
+    static readonly #encoder = new TextEncoder()
+
+    private static arrayToString(array: Uint8Array) {
+        return FileSystemCommon.#decoder.decode(array)
+    }
+
+    private static stringToArray(string: string): Uint8Array {
+        return FileSystemCommon.#encoder.encode(string)
+    }
+
+    private static asArray(array: Uint8Array | string): Uint8Array {
+        if (typeof array === 'string') {
+            return FileSystemCommon.stringToArray(array)
+        }
+        return array
+    }
+
+    /**
+     * Retrieve the Uri of the file.
+     *
+     * @param path The file path for which to retrieve metadata.
+     * @return The Uri about the file.
+     */
+    private static getUri(path: string | vscode.Uri): vscode.Uri {
+        if (path instanceof vscode.Uri) {
+            return path
+        }
+        return vscode.Uri.file(path)
+    }
+}
+
+const fsCommon = new FileSystemCommon()
+export default fsCommon
```

Comment:
We generally don't use `export default`. Is there a reason to introduce it? Can add friction to refactoring.

Reply:
Related to the instance getter, this code has been removed